design/metadata-handling.md

```diff
@@ -112,6 +117,11 @@ on Metal3Machine
 - Documentation of this feature would be very important.
 - The controllers must be able to recreate all status objects, i.e. no
   necessary information should be stored in the status without being recoverable
+- Since the BareMetalHost references the secrets and they are used at provisioning
+  time, the secrets cannot be updated to be able to reprovision the node in the
+  exact same state. This means that the data template parts containing the
+  metadata and network data must be immutable. Updates have to be done by creating
+  a new template and referencing it in the new / updated Metal3MachineTemplate.
```

Comment:
maybe it could be listed as an alternative that the M3DT could be mutable and reconciled so that the secrets would be updated, and reason why this would be problematic. For example, that the reprovisioning of the node would then lead to a different configuration, and that might be unexpected. And if we allow the mutability without updating the secrets,there is no way to know which version of the template they were created with and what they contain.

Reply:
Just to be clear, do you mean to say I should point out why M3DT mutability is not a viable option?

Reply:
yes, in the alternatives section

Reply:
Addressed